packages/core/ui-settings/core-ui-settings-server-internal/src/ui_settings_service.ts

```diff
@@ -148,6 +159,55 @@ export class UiSettingsService
     });
   };
 
+  private setAllowlist = (keys: string[]) => {
+    // Disabling this check for now since it causes some test failures
+    // if (this.allowlist) {
+    //   throw new Error(
+    //     `The uiSettings allowlist has already been set up. Instead of calling setAllowlist(), add your settings to packages/serverless/settings`
+    //   );
+    // }
+    this.allowlist = new Set(keys);
+  };
+
+  private validateAllowlist() {
+    this.allowlist?.forEach((key) => {
+      if (!this.uiSettingsDefaults.has(key) && !this.uiSettingsGlobalDefaults.has(key)) {
+        throw new Error(
+          `The uiSetting with key [${key}] is in the allowlist but is not registered. Make sure to remove it from the allowlist in /packages/serverless/settings`
```

Comment:
I was wondering if this is too strict. The default allow list is in a common package but the settings are registered by plugins. I wonder if any of those plugins can be disabled and then this throughs.

Reply:
This is a good point... We may want to apply the validation in dev mode only.

While `serverless` should be served as-is (no custom enabling/disabling of plugins per customer), and disabling plugins should be our last resort, as it could lead to unexpected and untested behavior, I wonder if we want to allow disabling plugins when there's a specific issue in prod that we want to mitigate.

cc @lukeelmers

Reply:
> This is a good point... We may want to apply the validation in dev mode only.

Sure, this makes sense to me. I added this condition with https://github.com/elastic/kibana/pull/164471/commits/6bfe3da1a07327c64d97ed9d7f6e6e4029efc62a